### DISCUSSION TITLE
Background Job Poll Interval time

### DISCUSSION THREAD

**Discussion by vsimic** [jobrunr/jobrunr#607] [Q&A]
In my property file I have this kind of configuration
`org:
  jobrunr:
    background-job-server:
      enabled: true
      worker-count: 1
      poll-interval-in-seconds: 30`
      
My understanding is that   background-job-server will have one instance of worker and that it will pool for anything in queue on 30s
Instead, I see that polling is done on 5s 

Am I missing something in the configuration?

**Comment by rdehuyss** [comment]
You must be overriding the pollIntervalInSeconds somewhere else as by default it is 15 and not 5.

**Reply by vsimic** [reply]
Thank you for your reply.

No, I have no additional configuration in place.
Bit more context:

We have around 100 entries added to queue to be processed by Background Server.
It looks like framework picks up next task to be processed/onboarded before poll interval finishes.
As mentioned, my understanding is that next Background Job will not be moved from enqueued to processing before polling interval end.

**Reply by rdehuyss** [reply]
Hhhm, no your understanding is wrong (and I may have given you wrong info in the past): as soon as the first job is done, JobRunr will immediately take the next job - even if the poll interval has not completed yet. This is to make sure that the throughput of jobs is as high as possible.

**Reply by vsimic** [reply]
Ahhh, that makes perfect sense though.

So if we would want such a behavior, we would need to create custom WorkDistributionStrategy implementation ?

Thank you for fast reply btw.
